CMakeLists.txt

```diff
@@ -71,12 +71,26 @@ include_directories(build/msgs)
 
 if (NOT freertps_standalone)
   ament_export_libraries(freertps freertps_system_${SYSTEM})
+
+  if (AMENT_ENABLE_TESTING)
+    find_package(ament_cmake_gtest REQUIRED)
+
+    ament_add_gtest(talker_listener_gtest "tests/talker_listener_test.cpp" TIMEOUT 30)
+    set_target_properties(talker_listener_gtest PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
I actually don't think this is necessary, I put it in as an experiment. I believe GCC will guess the linker language from the file extension.